src/server/dflycmd.cc

```diff
@@ -100,6 +100,32 @@ struct TransactionGuard {
   Transaction* t;
 };
 
+OpStatus WaitReplicaFlowToCatchup(absl::Time end_time, shared_ptr<DflyCmd::ReplicaInfo> replica,
```

Comment:
Why the `shared_ptr<DflyCmd::ReplicaInfo>` and not just a plain pointer? That way you don't need to call move two times when you call this and you get away with a single call to `.get()` function

Reply:
because a replica can disconnect